test/unit/run.test.js

```diff
@@ -70,4 +71,58 @@ describe('run module', function () {
             }
         ], done);
     });
+
+    it('must correctly resolve conflicts between iterationData.length and iterationCount', function (done) {
```

Comment:
By using nock to mock the API and on its reply add callback function with expect. 
https://www.npmjs.com/package/nock

But for now, simply add that it shouldn't throw non zero code and add a todo